### PR TITLE
Update Azure Storage Mgmt API and add CISv1.2.0 3.9 

### PR DESCRIPTION
**Pull request by xnkevinnguyen**
# Description
Do not merge

- Updgrade Azure Storage Mgmt API to latest version (16.0.0). https://pypi.org/project/azure-mgmt-storage/

- Add Rule 3.9 from Azure CISv1.2.0 Benchmark: 3.9 Ensure storage for critical data are encrypted with Customer Managed Key (Automated)

## Type of change

Select the relevant option(s):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by j4v** [comment]
> Do not merge

I've set it to `Draft` status :smile:

File: ScoutSuite/providers/azure/rules/rulesets/default.json [hunk 1]
```diff
@@ -291,6 +291,12 @@
                 "level": "warning"
             }
         ],
+        "storageaccount-encrypted-not-customer-managed.json": [
+            {
+                "enabled": true,
+                "level": "danger"
```
Comment (j4v) [inline ScoutSuite/providers/azure/rules/rulesets/default.json:297]:
```suggestion
                "level": "warning"
```

**Comment by j4v** [comment]
LGTM

File: ScoutSuite/providers/azure/rules/rulesets/default.json (hunk 1 above)
Comment (xnkevinnguyen) [reply ScoutSuite/providers/azure/rules/rulesets/default.json:297]: According to the CISv1.2.0 documentation, this rule has a profile applicability of level 2. Shouldn't this translate to a level of "Danger"?

**Pull request merged**